Automattic/jetpack
# Admin Menu: Remove "Links" menu

Fixes https://github.com/Automattic/wp-calypso/issues/48177

#### Changes proposed in this Pull Request:
Removes the "Links Manager" menu from the new unified menu since its usage in Core is [highly discouraged](https://core.trac.wordpress.org/ticket/21307#comment:73) and was deprecated very long ago. 

#### Jetpack product discussion
N/A.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Apply D58364-code to your WP.com sandbox.
* Sandbox the API and a WP.com simple site.
* Load that simple site in either Calypso or WP Admin.
* Make sure the "Links" menu is gone.
* Make sure you can still access it directly via `/wp-admin/link-manager.php`.
* Install Jetpack beta on a WoA site and switch to the branch of this PR.
* Load that WoA site in either Calypso or WP Admin.
* Make sure the "Links" menu is gone.
* Make sure you can still access it directly via `/wp-admin/link-manager.php`.
